extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/GrpcServerRecorder.java

```diff
@@ -706,6 +707,23 @@ public void stop(Promise<Void> stopPromise) {
                 }
             }
         }
+
+        @Override
+        public void beforeCheckpoint(org.crac.Context<? extends Resource> context) throws Exception {
+            Promise<Void> p = Promise.promise();
+            stop(p);
+            p.future().toCompletionStage().toCompletableFuture().get();
+        }
+
+        @Override
+        public void afterRestore(org.crac.Context<? extends Resource> context) throws Exception {
+            Promise<Void> p = Promise.promise();
+            // The verticle must be started by the event-loop thread; the thread calling
+            // afterRestore will likely do so for all suspended verticles, and had we called
+            // this directly the verticles would all share the same context (run on the same thread).
+            this.context.runOnContext(nil -> start(p));
```

Comment:
How can you be srue that this will start the verticles from the right event loop, using the event loop round robin and so on? 

I don't believe it's the right place to do that, but you should instrument how to undeploy and deploy verticles.